relay-sampling/src/dsc.rs

```diff
@@ -145,17 +145,27 @@ mod sample_rate_as_string {
 
     use serde::{Deserialize, Serialize};
 
+    #[derive(Debug, Clone)]
+    enum StringOrFloat<'a> {
+        String(Cow<'a, str>),
+        Float(f64),
+    }
```

Comment:
The custom deserializer and visitor is not necessary if you use `#[serde(untagged)]`, but I agree with Joris here, we can skip the entire type by just falling back to a float in the error case (that's what the serde macro would also generate).

Reply:
I did look into `#[serde(untagged)]` and from what I understand it is significantly slower to use it compared to a custom visitor.
Although in our case it might not have mattered much as we expect Strings anyway most of the time

Reply:
Did you benchmark it? Did you benchmark it against `deserialize_any` or just a specialized `deserialize_*`?

Without benchmarking or a really good hunch/reason why to prefer a custom deserializer + visitor, there is no reason to prefer this over the macro generated code here, it reduces the potential of bugs and code (+tests) required a lot.

But since we already have a deserializer we can do what the macro does internally and just handle a deserialization error as Joris suggested.

Reply:
I saw an issue describing the slow behaviour and there is documentation (albeit not in main) which "warns" about potential negative performance impact (https://github.com/epage/serde-rs.github.io/blob/untagged/_src/container-attrs.md#serdeuntagged-untagged).

I agree that it might have been a bit too eager from my side to use the custom visitor as I did not benchmark it myself